processing/src/main/java/io/druid/query/extraction/LookupExtractor.java

```diff
@@ -40,6 +41,18 @@
   @Nullable String apply(@NotNull String key);
 
   /**
+   * Provide the reverse mapping from a given value to a list of keys
+   * @param value the value to apply the reverse lookup
+   *              Null and empty are considered to be the same value = nullToEmpty(value)
+   *
+   * @return the list of keys that maps to value or empty list.
+   * Note that for the case of a none existing value in the lookup we have to cases either return an empty list OR list with null element.
+   * returning an empty list implies that user want to ignore such a lookup value.
+   * In the other hand returning a list with the null element implies user want to map the none existing value to the key null.
+   */
+  List<String> unApply(String value);
```

Comment:
do we have any safety checks built in for the case where the reverse lookup uses a significant amount of memory and will begin to impact performance?